src/templates/tournaments/tournament_detail.html

```diff
@@ -0,0 +1,70 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <h1>{{ tournament.name }}</h1>
+  <p>Game: {{ tournament.game }}</p>
+  <p>Start date: {{ tournament.start_date }}</p>
+  <p>End date: {{ tournament.end_date }}</p>
+  <p>Max players: {{ tournament.max_players }}</p>
+  <p>Description: {{ tournament.description }}</p>
+  <p>Rules: {{ tournament.rules }}</p>
+  <p>Draw rules: {{ tournament.draw_rules }}</p>
+  <p>Players:</p>
+  <ul>
+    {% for player in tournament.get_all_players %}
+      <li>
+        <p>
+          {% if player.name %}
+            <a href="{% url 'users:detail' player.id %}">{{ player.name }}</a>
+          {% else %}
+            <a href="{% url 'users:detail' player.id %}">{{ player.email }}</a>
```

Comment:
Possibly a data breach because email is visible to anyone and not just staff. Also there might be concerns with users who have made their accounts through API and don't have associated users. I'm okay with this for now, but what do you guys think? (You could also just leave this as something for future iterations of the class to deal with).

Reply:
Hi Elizabeth,
I think the change can be dealt with later because currently the user only need to provide an email when they create an account on the website (which means that the username would just be blank). As a result, if we do not have email displayed here, and the users have not setup their usernames, the way that a user is displayed would be a bunch of blank spaces, which is not ideal.

Reply:
Thank you for the review! The branch has been updated. For the changes in migration pages, I am not sure what causes the difference, but they should have the same content?

Reply:
Okay I'm okay with that decision! For the migration pages, I'm not really sure what's going on or why they are different, but could you just remove them from the PR adds?